src/coffeescript.coffee

```diff
@@ -186,7 +165,7 @@ exports.compile = compile = withPrettyErrors (code, options = {}) ->
   if options.inlineMap
     encoded = base64encode JSON.stringify v3SourceMap
     sourceMapDataURI = "//# sourceMappingURL=data:application/json;base64,#{encoded}"
-    sourceURL = "//# sourceURL=#{options.filename ? 'coffeescript'}"
+    sourceURL = "//# sourceURL=#{filename}"
     js = "#{js}\n#{sourceMapDataURI}\n#{sourceURL}"
```

Comment:
FYI, not as part of this PR but related: https://github.com/jashkenas/coffeescript/pull/5237